src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -440,8 +440,19 @@ static void restore_eliminated_locks(JavaThread* thread, GrowableArray<compiledV
     }
   }
 #ifdef ASSERT
-  if (LockingMode == LM_LIGHTWEIGHT && !realloc_failures) {
```

Comment:
In the new code, you are no longer account for `realloc_failures` being true.
I'm not convinced that is okay here.

Reply:
Originally I had those conditions in as well (to make it more clear and explicit). But Vladimir thought them superfluous. They are implicitly true from `lock_order.is_nonempty() -> LockingMode == LM_LIGHTWEIGHT && !realloc_failures` because we only ever add elements to `lock_order` if `LockingMode == LM_LIGHTWEIGHT && !realloc_failures` is true.

Reply:
Thanks for the clarification.